network-mux/src/Network/Mux.hs

```diff
@@ -220,7 +220,11 @@ miniProtocolJob tracer egressQueue
         writeTVar miniProtocolStatusVar StatusIdle
         putTMVar completionVar $ Right result
         buf <- readTVar miniProtocolIngressQueue
-        writeTVar miniProtocolIngressQueue $ BL.append remainder buf
+        case remainder of
+          Just trailing ->
+            writeTVar miniProtocolIngressQueue $ BL.append trailing buf
```

Comment:
I think you should move the reading of `buf` to inside the `Just` case.

Reply:
Yes; even better just use `modifyTVar`, we use strict `TVar`'s so this should be ok.